target/linux/brcm63xx/image/bcm63xx.mk

```diff
@@ -16,19 +16,23 @@ define Device/bcm63xx
   KERNEL := kernel-bin | append-dtb | relocate-kernel | lzma | lzma-cfe
   KERNEL_INITRAMFS := kernel-bin | append-dtb | lzma | loader-lzma elf
   IMAGES := cfe.bin
-  IMAGE/cfe.bin := cfe-bin --pad $$$$(shell expr $$$$(FLASH_SIZE) / 2)
-  IMAGE/cfe-4M.bin := cfe-bin --pad 2
-  IMAGE/cfe-8M.bin := cfe-bin --pad 4
-  IMAGE/cfe-16M.bin := cfe-bin --pad 8
+  IMAGE/cfe.bin := cfe-bin --pad $$$$(shell expr $$$$(FLASH_SIZE) / 2) | check-size $$$$(IMAGE_SIZE)
+  IMAGE/cfe-4M.bin := cfe-bin --pad 2 | check-size 0x3E0000
+  IMAGE/cfe-8M.bin := cfe-bin --pad 4 | check-size 0x7E0000
+  IMAGE/cfe-16M.bin := cfe-bin --pad 8 | check-size 0xFE0000
   IMAGE/cfe-bc221.bin := cfe-bin --layoutver 5
   IMAGE/cfe-old.bin := cfe-old-bin
-  IMAGE/sysupgrade.bin := cfe-bin
+  IMAGE/sysupgrade.bin := cfe-bin | check-size $$$$(IMAGE_SIZE)
+  BLOCK_SIZE := 0x10000
   FLASH_SIZE := 4
+# IMAGE_SIZE := $$(shell expr $$(FLASH_SIZE) \* 1024 \* 1024 - 2 \* $$(BLOCK_SIZE))
+# IMAGE_SIZE := $$(shell printf '$$(FLASH_SIZE) * 1024 * 1024 - 2 * %d\n' $$(BLOCK_SIZE) | bc)
+  IMAGE_SIZE := $$(shell expr $$(FLASH_SIZE) \* 1024 \* 1024 - 2 \* $$$$(printf '%d' $$(BLOCK_SIZE)))
   CFE_BOARD_ID :=
   CFE_CHIP_ID :=
-  CFE_EXTRAS :=
+  CFE_EXTRAS := --block-size $$(BLOCK_SIZE)
```

Comment:
How about also adding a IMAGE_OFFSET and doing `--image-offset $$(if $$(IMAGE_OFFSET),$$(IMAGE_OFFSET),$$(BLOCK_SIZE))`?
This would eliminate the need for modifying `CFE_EXTRAS` for most devices except those where it differs.